app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt

```diff
@@ -134,25 +144,54 @@ class MainActivity : AppCompatActivity() {
 
         @SuppressLint("SetTextI18n")
         override fun onUpdateCompleted() {
-            val frameStatistics = VisionManager.getFrameStatistics()
             runOnUiThread {
-                when (appModelPerformanceConfig) {
-                    is ModelPerformanceConfig.Merged -> {
-                        segmentation_fps.text = "S: 0"
-                        detection_fps.text = "D: 0"
-                        merge_model_fps.text = "MM: ${frameStatistics.segmentationDetectionFPS}"
-                    }
-                    is ModelPerformanceConfig.Separate -> {
-                        segmentation_fps.text = "S: ${frameStatistics.segmentationFPS}"
-                        detection_fps.text = "D: ${frameStatistics.detectionFPS}"
-                        merge_model_fps.text = "MM: 0"
+                with(VisionManager.getFrameStatistics()) {
+                    when (appModelPerformanceConfig) {
+                        is ModelPerformanceConfig.Merged -> {
+                            segmentation_fps.hide()
+                            detection_fps.hide()
+                            merge_model_fps.show()
+
+                            sumSegmentationDetectionFPS += segmentationDetectionFPS
+                            merge_model_fps.text =
+                                "MM: ${segmentationDetectionFPS.round()}  AVG: ${(sumSegmentationDetectionFPS / ++countSegmentationDetectionFPS).round()}"
+                        }
+                        is ModelPerformanceConfig.Separate -> {
+                            segmentation_fps.show()
+                            detection_fps.show()
+                            merge_model_fps.hide()
+
+                            sumSegmentationFPS += segmentationFPS
+                            sumDetectionFPS += detectionFPS
+                            segmentation_fps.text =
+                                "S: ${segmentationFPS.round()}  AVG: ${(sumSegmentationFPS / ++countSegmentationFPS).round()}"
+                            detection_fps.text =
+                                "D: ${detectionFPS.round()}  AVG: ${(sumDetectionFPS / ++countDetectionFPS).round()}"
+                        }
                     }
+
+                    sumCoreUpdatesFPS += coreUpdateFPS
+                    core_update_fps.text =
+                        "CU: ${coreUpdateFPS.round()}  AVG: ${(sumCoreUpdatesFPS / ++countCoreUpdatesFPS).round()}"
                 }
-                core_update_fps.text = "CU: ${frameStatistics.coreUpdateFPS}"
             }
         }
     }
 
+    private fun resetAverageFPS() {
```

Comment:
```suggestion
    private fun resetAverageFps() {
```

Reply:
FPS postfix is used for all properties, it would be better to use `...Fps` or `...FPS`, but it should be the same.
what do you think? 
@RingerJK 
@yunikkk

Reply:
I'd go with Fps

Reply:
Fps